xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -900,7 +900,10 @@ bool CGUIMediaWindow::OnClick(int iItem, const std::string &player)
     else if (!CProfilesManager::GetInstance().GetCurrentProfile().canWriteSources() && !g_passwordManager.IsProfileLockUnlocked())
       return false;
 
-    return OnAddMediaSource();
+    if (OnAddMediaSource())
+      Refresh();
+
+    return true;
```

Comment:
Should it not return the real result ? (BTW thanks for fixing this irritating bug)